api4/plugin_test.go

```diff
@@ -90,6 +91,11 @@ func TestPlugin(t *testing.T) {
 			}))
 			defer func() { slowTestServer.Close() }()
 
+			storeMock := mocks.Store{}
```

Comment:
@agnivade  I am mocking store and status again in few places which are giving errors so is this not what you meant by mocking again from all callers?

Reply:
Did you mean that `TestPlugin` is already failing for you? It passes for me though.

Anyways, looks like my latest suggestion did not work. I will create a separate branch and look into it to avoid having to ask you every time for changes.

Thanks for your patience on this @darkLord19